crates/starknet-devnet/tests/test_dump_and_load.rs

```diff
@@ -100,6 +100,49 @@ mod dump_and_load_tests {
         assert!(devnet_dump.is_err());
     }
 
+    #[tokio::test]
+    async fn dump_and_load_blocks_on_demand() {
+        let modes = vec!["exit", "block"];
+
+        for mode in modes {
+            let dump_file = UniqueAutoDeletableFile::new(
+                ("dump_load_dump_load_on_".to_owned() + mode).as_str(),
+            );
+
+            for _ in 0..2 {
+                let devnet_dump = BackgroundDevnet::spawn_with_additional_args(&[
+                    "--dump-path",
+                    &dump_file.path,
+                    "--dump-on",
+                    mode,
+                    "--blocks-on-demand",
+                ])
+                .await
+                .expect("Could not start Devnet");
+
+                devnet_dump.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
```

Comment:
Are there any blocks-on-demand tests that produce transactions that aren't minting? It's crucial for testing IMO

Reply:
I can introduce test like this

Reply:
Added in 414a3adb369ff7b50388c43fa457ec5fed3d4814